### PR TITLE
[MLIR][ONNX] Add OnnxToTorch support for MaxRoiPool Op

### PR DESCRIPTION
**Pull request by suryajasper**
This PR adds OnnxToTorch support for MaxRoiPool op

### CODE REVIEW

**Comment by vivekkhandelwal1** [comment]
Hi @suryajasper, can you please format the code as per the CI failures?

**Comment by suryajasper** [comment]
Hi @vivekkhandelwal1, formatting issues are resolved

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]
Thanks for adding the patch @suryajasper. You have written very clean code. I hope you've verified the e2e correctness of this code through SHARK-Testsuite. Please address the minor comments.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp [hunk 1]
```diff
@@ -613,6 +613,242 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
...
+        auto torchFloatTy = rewriter.getType<Torch::FloatType>();
+
+        Value spatialScaleValue = rewriter.create<Torch::ConstantFloatOp>(
+            loc, torchFloatTy, rewriter.getF64FloatAttr(spatialScale));
+
+        Value boolTrue = rewriter.create<Torch::ConstantBoolOp>(
+            loc, rewriter.getBoolAttr(true));
+
+        auto roisTy = rewriter.getType<Torch::ValueTensorType>(
+            roisFloatTy.getSizes(), floatTy);
```
[the remark this answers is not in the document]
Comment (suryajasper) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp]: I was able to remove a few redundancies here, but the remaining type variables in the code are necessary as the design of the ONNX operation requires a lot of type and shape manipulation of the input tensors.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp [hunk 2]
```diff
@@ -613,6 +613,242 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
...
+        }
+
+        Value outputShapeList =
+            createConstantIntList(binder, rewriter, pooledShape);
+
+        Torch::ValueTensorType resultTy;
+        Value input, rois;
+        if (binder.tensorOperands(input, rois) ||
+            binder.tensorResultType(resultTy))
+          return failure();
```
[the remark this answers is not in the document]
Comment (suryajasper) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp]: Fixed!

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp [hunk 3]
```diff
@@ -613,6 +613,242 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
...
+          return rewriter.notifyMatchFailure(
+              binder.op, "Rank of input tensor must be >= 4");
+        }
+
+        ArrayRef<int64_t> roisShape = roisTy.getSizes();
+        if (roisShape.size() != 2 || roisShape[1] != 5) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected ROIs to be tensor of rank 2: (num_rois, 5)");
+        }
+        int64_t numRois = roisShape[0];
```
[the remark this answers is not in the document]
Comment (suryajasper) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp:669]: Fixed!

**Review by vivekkhandelwal1 — APPROVED** [review]

**Pull request merged**